targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_WIO_3G/onboard_modem_api.c

```diff
@@ -21,50 +23,62 @@
 #include "platform/mbed_wait_api.h"
 #include "PinNames.h"
 
+#define WAIT_AFTER_POWR_CHANGED	(1000)	// [msec.]
+
 #if MODEM_ON_BOARD
 
 static void press_power_button(int time_ms)
 {
     gpio_t gpio;
 
     gpio_init_out_ex(&gpio, PWRKEY, 1);
-    gpio_write(&gpio, 1);
     wait_ms(time_ms);
     gpio_write(&gpio, 0);
 }
 
 void onboard_modem_init()
 {
     gpio_t gpio;
-    // start with modem disabled
+
+    // Power Supply
+    gpio_init_out_ex(&gpio, M_POWR, 0);
+    // Turn On/Off
+    gpio_init_out_ex(&gpio, PWRKEY, 0);
     gpio_init_out_ex(&gpio, RESET_MODULE, 0);
+    // Status Indication
     gpio_init_in_ex(&gpio, MDMSTAT, PullUp);
-    gpio_init_out_ex(&gpio, MDMDTR,       0);
-    gpio_init_out_ex(&gpio, M_POWR, 1);
+    // Main UART Interface
+    gpio_init_out_ex(&gpio, MDMDTR, 0);
 
-//    gpio_write(&gpio, M_POWR, 1);
-    wait_ms(500);
+    wait_ms(WAIT_AFTER_POWR_CHANGED);
 }
 
 void onboard_modem_deinit()
 {
-//    wio3g_mdm_powerOff();
+    onboard_modem_power_down();
```

Comment:
Should this be `gpio_init_out_ex(&gpio, M_POWR, 1);`, that's vice versa from `onboard_modem_init()`?

Reply:
We definitely need this to fix the issue #9450

Reply:
@ytsuboi

Reply:
I agree with you.  That makes more sense.  I will change this.

Reply:
@AriParkkila changed.